### PR TITLE
Feature/docker compose

### PR DESCRIPTION
**Pull request by ZaradarDFDS**
Containerized blaster

### CODE REVIEW

File: src/Blaster.WebApi/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#Setup build image
+FROM mcr.microsoft.com/dotnet/core/sdk:2.2 AS build-env
+
+#Set container filesystem to /build (and create folder if it doesnt exist)
+WORKDIR /build
+
+# install dependencies 
+RUN apt-get update && apt-get install -y apt-utils && apt-get install -y curl
+
+# get install script and pass it to execute:
```
Comment (kim-lindhard-dfds) [inline src/Blaster.WebApi/Dockerfile]: # get Node.js install script and pass it to execute

File: src/Blaster.WebApi/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#Setup final container images.
+FROM mcr.microsoft.com/dotnet/core/aspnet:2.2 AS runtime
+WORKDIR /app
+
+#Copy binaries from publish container to final container
+COPY --from=build-env /build/src/Blaster.WebApi/wwwroot ./wwwroot
+COPY --from=build-env /build/out .
+
+# SSL
+ENV CAPABILITY_SERVICE_KAFKA_SSL_CA_LOCATION=/app/cert.pem
```
Comment (kim-lindhard-dfds) [inline src/Blaster.WebApi/Dockerfile]: [nitpick] move to line 52. The lines 49 to 51 is about the aws cert not the kafka cert.

Comment (ZaradarDFDS) [reply src/Blaster.WebApi/Dockerfile]: moved :)

**Comment by ZaradarDFDS** [comment]
> First impressions so far:
> 
> * With Rider, ensure it uses the docker-compose file instead of the Dockerfile that it might auto suggest.
> * There seems to be no hot reloading functionality like the current workflow(e.g. 'dotnet watch' and 'webpack --watch --hot --mode development'). This already slowed down just my testing, I shudder to think about how it'll be for actual development. Having to stop and start the compose again manually is rather cumbersome not to mention quite slow on my machine (i7-8650U, 16GB 2400Mhz).
> 
> With the above comments in mind, if this is to completely replace the current development workflow, I can't wholeheartedly approve this PR. However, if the previous workflow (using the scripts in local-development, which currently is broken due to changes with the docker-compose.yml in fake_dependencies. I guess one could maintain a separate Compose file for that) is still going to be there, I'm fine with this.

Regarding support for Rider and debugging docker-compose that is outside the scope of this pull request and possible this repo. More info about Rider support for docker-compose can be found @ https://youtrack.jetbrains.com/issue/RIDER-17801?_ga=2.263162632.502084918.1573567200-1230062943.1570100500&_gac=1.56032601.1570100501.Cj0KCQjwuNbsBRC-ARIsAAzITucpxtgwFDaHZADsXWa4XPI4JWXHskIwfzyKBEqBgU0AEjwG_xYUKs0aApjvEALw_wcB

With regards to hot reloading I have spent the better part of the a day trying to get this to work in an acceptable manner for all parties and my current conclusion is that the value we gain from rigging this together does in no way outweight the drawsbacks of bloating our final image with volumes and shell scripts to launch multiple processes. Especially considering that this requirement is for "development environments" only and not being able to "hot reload" does not prevent development environments from working, it simply optimizes the development experience (something that is better done without docker anyways if you have multiple "rebuilds").

**Comment by ZaradarDFDS** [comment]
We now have full support for docker-compose (with hot-reloading) and can retire pipeline.sh in favor of a cross-platform alternative that will enable us to debug our local dev in the same context it will be running in prod (containerized). :) All with out bloating the image that goes into production. Yaya!

**Comment by ZaradarDFDS** [comment]
> Works just fine by itself. With Rider it's a bit of a pain the arse.

Use a better IDE :D

**Pull request merged**